src/sql/workbench/api/common/notebooks/vscodeExecuteProvider.ts

```diff
@@ -221,25 +231,23 @@ class VSCodeSessionManager implements azdata.nb.SessionManager {
 	}
 
 	public get specs(): azdata.nb.IAllKernels {
-		let languages = this._controller.supportedLanguages?.length > 0 ? this._controller.supportedLanguages : [this._controller.label];
+		let defaultKernel: azdata.nb.IKernelSpec = {
+			name: this._controller.notebookType,
+			language: this._controller.supportedLanguages[0],
+			display_name: this._controller.label,
+			supportedLanguages: this._controller.supportedLanguages ?? []
+		};
 		return {
-			defaultKernel: languages[0],
-			kernels: languages.map<azdata.nb.IKernelSpec>(language => {
-				return {
-					name: language,
-					language: language,
-					display_name: language
-				};
-			})
+			defaultKernel: defaultKernel.name,
+			kernels: [defaultKernel]
```

Comment:
Why are you only returning the default kernel now? Would suggest adding a comment about that to the code

Reply:
These specs get accessed before the kernels themselves are created, so we don't have any more detailed information to return here. Trying to append the kernel information later also didn't help, since it seems like our code assumes kernel specs will always remain constant, and so never bothers to update kernelSpec info after that initial query.